src/views/shared/amount-input/amount-input.view.jsx

```diff
@@ -92,12 +94,17 @@ function AmountInput(Component) {
      * @param {BigNumber} newAmount - New amount to be checked.
      */
     function checkAmountValidity(newAmount) {
-      const fee = getTransactionFee(transactionType, newAmount, account.token, l2Fee, gasPrice);
-      const newAmountWithFee = newAmount.add(fee);
+      const newFee = getTransactionFee(transactionType, newAmount, account.token, fee, gasPrice);
+      const newAmountWithFee = newAmount.add(newFee);
+      const isNewAmountMoreThanFunds = newAmountWithFee.gt(BigNumber.from(account.balance));
 
       setIsAmountNegative(newAmountWithFee.lte(BigNumber.from(0)));
-      setIsAmountMoreThanFunds(newAmountWithFee.gt(BigNumber.from(account.balance)));
-
+      setIsAmountMoreThanFunds(isNewAmountMoreThanFunds);
+      setIsAmountMoreThanWithFees(
+        isNewAmountMoreThanFunds &&
+          transactionType === TxType.Deposit &&
```

Comment:
This show an alert on the parent component and send a signal that the amount is invalid and  `isAmountLessThanFundsButMoreWithFees` 
The behaivour of `newAmountWithFee` is valid for every TxType

Reply:
I still don't get why this logic only applies to a `Deposit`. I mean, why we don't want to benefit from this feature when we are doing a withdraw or exit? Like here, where it "looks like" I do have enough funds, but still get an error because of the fee. Wouldn't it be nice to let the user know the fee is the reason as we do in the deposit?

![image](https://user-images.githubusercontent.com/15896177/135050118-7124efed-db9b-43fa-94d4-aee5bbf3503f.png)

Reply:
I think that this feature was requested just for deposit and `AmountInput` is  just taking in account the L2 fee in a exit Tx. For your approach we have something similar that is checking if you have enough eth amount in a exit
![image](https://user-images.githubusercontent.com/19190248/135058301-e45f0be4-4f85-4e36-93cf-a79127227b16.png)

Reply:
But we can remove that conditional because that logic just affect if the parent is listening `areFundsExceededDueToFee` and maybe in the future some new feature consume it

Reply:
mmm... interesting, then I don't get why I'm not seeing this msg in the capture above...

Ok, anyway, if this was planned only for Deposits I close this one. Thx!